### DISCUSSION TITLE
displaying json column with table and updating it

### DISCUSSION THREAD

**Discussion by altayevrim** [Laravel-Backpack/community-forum#971] [Q&A (Help)]
Hi,
I have a column named as "raw" which stores a JSON. Something like this
```
"{\"kod\":\"KOD\",\"unvan\":\"UNVAN  \\/UNVAN\",\"tarih\":\"45209\",\"aciklama\":\"ACIKLAMA\",\"borc\":\"0\",\"alacak\":\"0\",\"bakiye\":\"0\",\"kayit_no\":\"0\",\"fis\":\"FIS TIPI\",\"no\":\"0\",\"surucu\":\"SURUCU\"}"
```

When I use the code below on **setupShowOperation** it works perfectly. 

```
$this->crud->column([
            'name'     => 'raw',
            'label'    => 'Kayıt Detayları',
            'type'     => 'table',
            'columns' => [
                'kod'      => 'Kod',
                'unvan'    => 'Ünvan',
                'tarih'         => 'Tarih',
                'aciklama'      => 'Açıklama',
                'borc'          => 'Borç',
                'alacak'        => 'Alacak',
                'bakiye'        => 'Bakiye',
                'kayit_no'      => 'Kayıt No',
                'fis'      => 'Fiş',
                'no'        => 'No',
                'surucu'         => 'Sürücü'
            ]
        ],);
```
But when I use the field version of it on **setupUpdateOperation** it does not work at all.  The values don't get rendered.   
I need to enable editing as well, tried fake fields too but it didn't work either.   

What should I do?

**Comment by karandatwani92** [comment]
Hey @altayevrim 

I tested it, and it worked perfectly for me. Though it stores like this:
```
{
    "en": [
        {
            "no": "1",
            "fis": "1",
            "kod": "1",
            "borc": "1",
            "tarih": "1",
            "unvan": "1",
            "alacak": "1",
            "bakiye": "1",
            "surucu": "1",
            "aciklama": "1",
            "kayit_no": "1"
        }
    ]
}
```
![Screenshot 2024-05-01 at 5 08 56 PM](https://github.com/Laravel-Backpack/community-forum/assets/8214221/d0a19998-7089-4525-a7a3-13ee606a0273)

I suspect something is wrong with casting. In my model, i used the following:
```php
public $casts = [
   'raw'       => 'object',
]
```

**Reply by altayevrim** [reply]
Hi, I tried object casting as well. Didn't help unfortunately. Have you checked updating the values? Display worked fine for me too. I don't use it as translatable by the way. So storing is a bit different on my end.

**Reply by karandatwani92** [reply]
> ```
> {
>     "en": [
>         {
>             "no": "1",
>             "fis": "1",
>             "kod": "1",
>             "borc": "1",
>             "tarih": "1",
>             "unvan": "1",
>             "alacak": "1",
>             "bakiye": "1",
>             "surucu": "1",
>             "aciklama": "1",
>             "kayit_no": "1"
>         }
>     ]
> }
> ```

Yes, I updated using the form, and this was the stored value.

**Reply by altayevrim** [reply]
Thank you very much for your reply. Then something is odd in my environment. I am not using translations, might be the reason for it? Or I'm storing this JSON on my database. So that might be the reason why the update operation / form does not show the values properly while show operation shows without an issue. I'll take a look at it further.
```json
"{\"kod\":\"KOD\",\"unvan\":\"UNVAN  \\/UNVAN\",\"tarih\":\"45209\",\"aciklama\":\"ACIKLAMA\",\"borc\":\"0\",\"alacak\":\"0\",\"bakiye\":\"0\",\"kayit_no\":\"0\",\"fis\":\"FIS TIPI\",\"no\":\"0\",\"surucu\":\"SURUCU\"}"
```

**Reply by karandatwani92** [reply]
Your saved JSON seems double-encoded. Try my solution on a new/add item and check if it works on it. Do not edit the previous saved value.
